src/sentry/integrations/aws_lambda/utils.py

```diff
@@ -322,6 +346,10 @@ def wrapped(*args, **kwargs):
                 # only have one specific error to catch
                 if invalid_layer:
                     raise IntegrationError(_(INVALID_LAYER_TEXT) % invalid_layer)
+                # check if error raised is a missing role message
+                missing_role = get_missing_role_error(err_message)
+                if missing_role:
+                    raise IntegrationError(_(MISSING_ROLE_TEXT))
```

Comment:
@ahmedetefy This looks good but I believe we also need to update this block as well:

https://github.com/getsentry/sentry/blob/427d73d931ba258bb87717718d56981a8e2a2479/src/sentry/integrations/aws_lambda/integration.py#L382-L388

Would be nice if we could refactor this so when we have subsequent errors we only have to update one place but I'm not exactly sure the beset way of doing that.